arrow/parser.py

```diff
@@ -423,6 +423,21 @@ def _build_datetime(parts):
         elif am_pm == "am" and hour == 12:
             hour = 0
 
+        # Support for midnight at the end of day
+        if hour == 24:
+            if parts.get("minute"):
```

Comment:
We should change this and the following conditionals to `if parts.get("minute") is not None:` for readability purposes.

Reply:
@jadchaar I just made the change, but I had to check against `0` because for this condition is the same if the minute/second/microsecond does not exist or is equal to `0`

Reply:
Good catch!